record.go

```diff
@@ -197,6 +198,34 @@ func (c *Client) DeleteRecord(ctx context.Context, k *Key) error {
 // If there is no such record matching the given key this method returns an
 // error wrapping ErrNoSuchRecord.
 func (c *Client) GetRecord(ctx context.Context, k *Key) (Record, error) {
+	if c.v4 {
+		ctx = context.WithValue(ctx, openapi.ContextBasicAuth, c.openAPI.auth)
+
+		req := openapi.Sajariv4beta1GetRecordRequest{
+			Key: openapi.Sajariv4beta1Key{
+				Field: k.field,
+				Value: fmt.Sprintf("%v", k.value),
+			},
+		}
+		resp, _, err := c.openAPI.client.RecordsApi.GetRecord(ctx, c.Collection).Sajariv4beta1GetRecordRequest(req).Execute()
+		if err != nil {
+			switch x := err.(type) {
+			case openapi.GenericOpenAPIError:
+				m := x.Model()
+
+				if m, ok := m.(openapi.GatewayruntimeError3); ok {
+					switch codes.Code(m.GetCode()) {
+					case codes.NotFound:
+						return nil, fmt.Errorf("%v: %w", k, ErrNoSuchRecord)
+					}
+				}
+			}
+			return nil, fmt.Errorf("could not get record: %w", err)
+		}
+
+		return resp, nil
+	}
```

Comment:
I would maybe break this out into another method so this looks like
```
if c.v4 {
    return c.getRecordV4(ctx, k)
}

// ... rest of code
}
```

Reply:
@benhinchley done